### ISSUE TITLE
feature request: conditional transitions

### ISSUE THREAD

**Issue by donpdonp** [Bockit/state-transition-system#1]
In browsing a bunch of state machine gems, this one has a good API. One important feature that seems to be missing is having criteria for a state change. For example, the first function passed to addTransition could return true/false to allow the state change.

``` javascript
var car = new StateMachine('car')
car.addTransition('car','freeway') // initial state, 50MPH
car.addTransition('car', 'parked', function(){
  // only park when stopped
  return car.speed == 0
}, function() {
  transmission.setGear('PARK')
})

// this state change should not happen at 50MPH 
car.stateChange('parked')
```

**Comment by Bockit** [comment]
Sorry for the slow reply.

When it's blocked, should it throw an error or just not change?

EDIT: Or a better question: Do you need to know that a state change was blocked? I can imagine that you would want to in some scenarios.

**Comment by donpdonp** [comment]
The semantics get tricky because there are multiple valid ways to go about it.

To me, the most reasonable default is If any state transition function returns false, I'd say the state change does no happen, and emit a 'transitionblocked' message with the details of which state change was attempted, and which function returned false.

**Comment by Bockit** [comment]
Ahh yes, haha I forgot about the return value. That works, will make the changes

**Comment by Bockit** [comment]
Sorry, this dropped off my radar. Probably too late for you, but I've published (2.0.0) the changes allowing return false in a transition cancelling the state change.
